providers/implementations/skeymgmt/aes_skmgmt.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2025 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include "internal/deprecated.h"
+
+#include <openssl/core_dispatch.h>
+#include <openssl/core_names.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/proverr.h>
+#include "prov/implementations.h"
+#include "prov/providercommon.h"
+#include "prov/provider_ctx.h"
+#include "internal/skey.h"
+#include "crypto/types.h"
+#include "internal/param_build_set.h"
+
+static OSSL_FUNC_skeymgmt_import_fn aes_import;
+static OSSL_FUNC_skeymgmt_export_fn aes_export;
+static OSSL_FUNC_skeymgmt_free_fn aes_free;
+
+static void aes_free(void *keydata)
+{
+    SKEY *aes = keydata;
+
+    if (aes == NULL)
+        return;
+
+    OPENSSL_free(aes->data);
+    OPENSSL_free(aes);
+}
+
+static void *aes_import(void *provctx, int selection,
+                        const OSSL_PARAM params[])
+{
+    OSSL_LIB_CTX *libctx = PROV_LIBCTX_OF(provctx);
+    const OSSL_PARAM *raw_bytes;
+    SKEY *aes;
+    int ok = 1;
+
+    if (!ossl_prov_is_running())
+        return NULL;
+
+    if ((selection & OSSL_SKEYMGMT_SELECT_SECRET_KEY) == 0)
+        return NULL;
+
+    raw_bytes = OSSL_PARAM_locate_const(params, OSSL_SKEY_PARAM_RAW_BYTES);
+    if (raw_bytes == NULL)
+        return NULL;
+
+    aes = OPENSSL_zalloc(sizeof(SKEY));
+    aes->libctx = libctx;
+
+    aes->type = SKEY_TYPE_AES;
```

Comment:
It this is the only difference between the generic and aes specific versions then there has to be a better way to do this.

Reply:
Not the only difference, for example when importing an AES key we check for size and refuse incompatible ones.

Reply:
I have no issues with skey having a type to it. But having this sort of boiler plate code repeated for every symmetric key type seems like the wrong approach. At the cipher layer we have this sort of info. Why here?

Reply:
I reimplemented the AES SKEYMGMT via generic one with some extra checks